### PR TITLE
Letting clients know when auth is expired.

### PR DESCRIPTION
**Pull request by andrewconner**
For the extension, mobile, and the site, the endpoint that gives user data now will include a `'notAuthed': [array, of, social, networks, that, are, expired]`. Example: `'notAuthed':['facebook','linkedin']`

@joonho1101, @2is10 @ebfaed (You may not need this yet, but the site does so I included it everywhere since it's incoming for all clients)

### CODE REVIEW

**Comment by ebfio** [comment]
Cool! Thanks.

File: kifi-backend/modules/shoebox/app/com/keepit/model/SocialUserInfoRepo.scala [hunk 1]
```diff
@@ -73,6 +74,9 @@ class SocialUserInfoRepoImpl @Inject() (
       (for(f <- table if f.userId === userId) yield f).list
     }
 
+  def getNotAuthorizedByUser(userId: Id[User])(implicit session: RSession): Seq[SocialUserInfo] =
+      (for(f <- table if f.userId === userId && f.state === SocialUserInfoStates.APP_NOT_AUTHORIZED) yield f).list
```
Comment (2is10) [inline kifi-backend/modules/shoebox/app/com/keepit/model/SocialUserInfoRepo.scala:78]: This only includes networks that the user authorized at some time in the past, correct?

**Comment by 2is10** [comment]
Thanks.

We may want to migrate the site and mobile app to their own auth endpoints (not in ExtAuthController).

File: kifi-backend/modules/shoebox/app/com/keepit/model/SocialUserInfoRepo.scala (hunk 1 above)
Comment (andrewconner) [inline kifi-backend/modules/shoebox/app/com/keepit/model/SocialUserInfoRepo.scala:78]: Correct. It's not a prompt to auth new networks, it's a reminder that a network's token has expired.

Comment (2is10) [inline kifi-backend/modules/shoebox/app/com/keepit/model/SocialUserInfoRepo.scala:78]: Are we able to distinguish expiration from a user manually revoking authorization?

Comment (andrewconner) [inline kifi-backend/modules/shoebox/app/com/keepit/model/SocialUserInfoRepo.scala:78]: Not 100%. We've chosen to treat them the same though. The site will only show the user a reminder at the point of need (i.e., while they're trying to invite people). The action is the same for any case — send them to the provider to re-auth, and they'll come back.

Comment (2is10) [inline kifi-backend/modules/shoebox/app/com/keepit/model/SocialUserInfoRepo.scala:78]: I see. Thanks.

**Pull request merged**